lsm-tree.cabal

```diff
@@ -52,45 +52,103 @@ description:
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Insert\/Delete\/Update | @MergePolicyLazyLevelling@ | \(O(\log_T \: n)\)           |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Duplicate              | N\/A                       | \(O(1)\)                     |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Union                  | N\/A                       | \(O(n)\)                     |
+  +----------+------------------------+----------------------------+------------------------------+
+  | Snapshot | Save                   | N\/A                       | \(O(\log_T \: n)\)           |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Open                   | N\/A                       | \(O(n)\)                     |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Delete                 | N\/A                       | \(O(n)\)                     |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | List                   | N\/A                       | \(O(s)\)***                  |
+  +----------+------------------------+----------------------------+------------------------------+
+  | Cursor   | Create                 | N\/A                       | \(O(\log_T \: n)\)           |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Close                  | N\/A                       | \(O(\log_T \: n)\)           |
+  +----------+------------------------+----------------------------+------------------------------+
+  |          | Read next entry        | N\/A                       | \(O(1)\)                     |
+  +----------+------------------------+----------------------------+------------------------------+
+
+  (*The variable \(t\) refers to the number of open tables in the session. **The variable \(b\) refers to the number of entries retrieved by the range lookup. ***The variable \(s\) refers to the number of snapshots in the session.)
+
+  === In-memory size of tables #performance_size#
 
   The in-memory size of an LSM-tree is described in terms of the variable \(n\), which refers to the number of /physical/ database entries.
   A /physical/ database entry is any key–operation pair, e.g., @Insert k v@ or @Delete k@, whereas a /logical/ database entry is determined by all physical entries with the same key.
 
   The worst-case in-memory size of an LSM-tree is \(O(n)\).
 
-  *   The worst-case size of the write buffer is \(O(1)\).
+  *   The worst-case in-memory size of the write buffer is \(O(1)\).
```

Comment:
I wonder if we can really consider the write buffer size a constant. It's a parameter to the table, no? What are your thoughts about this?

Reply:
It's a constant, because it's a parameter. We can use `N` to represent it, but in most cases it gets absorbed by the O.